### PR TITLE
[FLINK-36094] CDC SchemaRegistryRequestHandler should throw exception which is not SchemaEvolveException

### PR DESCRIPTION
**Pull request by loserwang1024**
As shown in https://issues.apache.org/jira/browse/FLINK-36094, Current, SchemaRegistryRequestHandler only throw 
SchemaEvolveException, which will not handle the others(like network, oom, or else.

### CODE REVIEW

**Comment by loserwang1024** [comment]
@leonardBang @yuxiqian @ruanhang1993 , CC, WDYT?

File: flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/schema/SchemaEvolveTest.java [hunk 1]
```diff
@@ -1040,8 +1040,10 @@ tableId, buildRecord(INT, 2, STRING, "Bob", SMALLINT, (short) 18)),
                                                 Column.physicalColumn(
                                                         "height", DOUBLE, "Height data")))));
         Assertions.assertThatThrownBy(() -> processEvent(schemaOperator, addColumnEvents))
+                .cause()
+                .cause()
                 .isExactlyInstanceOf(RuntimeException.class)
-                .hasMessageContaining("Failed to apply schema change");
+                .hasMessageContaining("failed to apply schema change");
```
Comment (yuxiqian) [inline flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/schema/SchemaEvolveTest.java:1046]: Use `hasRootCauseInstanceOf` and `hasRootCauseMessage`

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/org/apache/flink/cdc/connectors/doris/sink/DorisMetadataApplier.java [hunk 2]
```diff
@@ -221,7 +221,7 @@ private void applyAddColumnEvent(AddColumnEvent event) throws SchemaEvolveExcept
                         tableId.getSchemaName(), tableId.getTableName(), addFieldSchema);
             }
         } catch (Exception e) {
-            throw new SchemaEvolveException(event, e.getMessage(), e);
+            throw new SchemaEvolveException(event, "fail to apply add column event", e);
```
Comment (yuxiqian) [inline flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/org/apache/flink/cdc/connectors/doris/sink/DorisMetadataApplier.java:224]:
```suggestion
            throw new SchemaEvolveException(event, "Failed to apply add column event", e);
```

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java [hunk 3]
```diff
@@ -437,6 +444,25 @@ private List<SchemaChangeEvent> lenientizeSchemaChangeEvent(SchemaChangeEvent ev
         }
     }
 
+    private boolean shouldIgnoreException(Exception exception) {
+
+        // only UnsupportedSchemaChangeEventException maybe ignore(depends on SchemaChangeBehavior)
```
Comment (yuxiqian) [inline flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java:449]: Code flow and comments are confusing here. Something like

```java
// Only `UnsupportedSchemaChangeEventException` could be ignored in TRY_EVOLVE mode.
// In IGNORE mode, will never try to apply schema change events
// In EVOLVE mode, such failure will not be tolerated
// In EXCEPTION mode, an exception will be thrown once captured
return (exception instanceof UnsupportedSchemaChangeEventException) && 
    (schemaChangeBehavior == TRY_EVOLVE)
```

might be enough.

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java [hunk 4]
```diff
@@ -147,17 +146,27 @@ private void applySchemaChange(
                 try {
                     metadataApplier.applySchemaChange(changeEvent);
                     LOG.debug("Applied schema change {} to table {}.", changeEvent, tableId);
+                    schemaManager.applyEvolvedSchemaChange(changeEvent);
                     finishedSchemaChanges.add(changeEvent);
-                } catch (SchemaEvolveException e) {
+                } catch (Exception e) {
```
Comment (ruanhang1993) [inline flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java:151]:
```suggestion
                } catch (Throwable t) {
```

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java (hunk 3 above)
Comment (loserwang1024) [reply flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java]: done it

File: flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/org/apache/flink/cdc/connectors/doris/sink/DorisMetadataApplier.java (hunk 2 above)
Comment (loserwang1024) [reply flink-cdc-connect/flink-cdc-pipeline-connectors/flink-cdc-pipeline-connector-doris/src/main/java/org/apache/flink/cdc/connectors/doris/sink/DorisMetadataApplier.java:224]: done it

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java (hunk 4 above)
Comment (loserwang1024) [reply flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistryRequestHandler.java]: done it

**Comment by loserwang1024** [comment]
> Could you please modify these parts together?

done it

**Review by yuxiqian — APPROVED** [review]
Thanks for @loserwang1024's rapid response, LGTM

File: flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/schema/SchemaEvolveTest.java [hunk 5]
```diff
@@ -1040,6 +1040,8 @@ tableId, buildRecord(INT, 2, STRING, "Bob", SMALLINT, (short) 18)),
                                                 Column.physicalColumn(
                                                         "height", DOUBLE, "Height data")))));
         Assertions.assertThatThrownBy(() -> processEvent(schemaOperator, addColumnEvents))
+                .cause()
+                .cause()
```
[the remark this answers is not in the document]
Comment (loserwang1024) [reply flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/schema/SchemaEvolveTest.java:1044]: I have tried it, but the rootCause is akka error(which can not show the real failed reason in test.

[the remark this answers is not in the document]
Comment (ruanhang1993) [reply flink-cdc-runtime/src/test/java/org/apache/flink/cdc/runtime/operators/schema/SchemaEvolveTest.java:1044]: OK. I get it.

File: flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistry.java [hunk 6]
```diff
@@ -275,6 +277,8 @@ public void resetToCheckpoint(long checkpointId, @Nullable byte[] checkpointData
                     throw new IOException(
                             "Unrecognized serialization version " + schemaManagerSerializerVersion);
             }
+        } catch (Throwable throwable) {
+            context.failJob(throwable);
```
[the remark this answers is not in the document]
Comment (loserwang1024) [reply flink-cdc-runtime/src/main/java/org/apache/flink/cdc/runtime/operators/schema/coordinator/SchemaRegistry.java]: add it now

File: flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/utils/PipelineTestEnvironment.java [hunk 7]
```diff
@@ -88,7 +88,7 @@ public abstract class PipelineTestEnvironment extends TestLogger {
 
     @Parameterized.Parameters(name = "flinkVersion: {0}")
     public static List<String> getFlinkVersion() {
-        return Arrays.asList("1.17.2", "1.18.1", "1.19.1", "1.20.0");
+        return Arrays.asList("1.17.2");
```
Comment (ruanhang1993) [inline flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/utils/PipelineTestEnvironment.java:91]: Why delete other flink versions here?

Comment (loserwang1024) [reply flink-cdc-e2e-tests/flink-cdc-pipeline-e2e-tests/src/test/java/org/apache/flink/cdc/pipeline/tests/utils/PipelineTestEnvironment.java:91]: Sorry, I do it for test and will modify back later.

**Review by leonardBang — APPROVED** [review]
Thanks @loserwang1024 for the improvement and @yuxiqian for the review +1

**Pull request merged**